### PR TITLE
Minor code cleanup in PackageSigning

### PR DESCRIPTION
**Pull request by yim-lee**
- Remove macOS 11 check since macOS 12 is required now
- Add test

### CODE REVIEW

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Review by neonichu — APPROVED** [review]

**Review by MaxDesiatov — APPROVED** [review]

File: Tests/PackageSigningTests/SigningTests.swift [hunk 1]
```diff
@@ -648,6 +648,69 @@ final class SigningTests: XCTestCase {
...
+            observabilityScope: ObservabilitySystem.NOOP
+        )
+
+        guard case .valid = status else {
+            return XCTFail("Expected signature status to be .valid but got \(status)")
+        }
+
+        func ecADPKeyAndCertChain(callback: (Result<KeyAndCertChain, Error>) -> Void) {
+            do {
+                try fixture(name: "Signing", createGitRepo: false) { fixturePath in
```
Comment (MaxDesiatov) [inline Tests/PackageSigningTests/SigningTests.swift:693]: Is the fact that `fixture` is callback-based the only reason to make `ecADPKeyAndCertChain` callback-based?

Comment (yim-lee) [reply Tests/PackageSigningTests/SigningTests.swift:693]: Yes. Is there an async version of `fixture`?

Comment (tomerd) [reply Tests/PackageSigningTests/SigningTests.swift:693]: we should make one!

**Pull request merged**